src/Symfony/Component/Cache/Adapter/AbstractAdapter.php

```diff
@@ -406,6 +437,29 @@ protected static function unserialize($value)
         }
     }
 
+    /**
+     * Increments a value in the cache by the given step value and returns the new value.
+     *
+     * If the key does not exist, it is initialized to the value of $step.
+     *
+     * @param string $key  The cache item key
+     * @param int    $step The value to increment by
+     *
+     * @return int|bool The new value on success and false on failure
+     */
+    protected function doIncrement($id, $step)
+    {
+        foreach ($this->doFetch(array($id)) as $value) {
+            if (is_numeric($value)) {
+                $step += $value;
+            }
+        }
+
+        $e = $this->doSave(array($id => $step), 0);
```

Comment:
this implementation in the base adapter is suspicious. The spec asks for atomicity, but atomicity requires to overwrite this method in each child adapter. An adapter not overwriting it would not provide atomicity, thus breaking the spec.